crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -749,18 +749,23 @@ impl AuthorityEpochTables {
 
     pub fn write_transaction_locks(
         &self,
-        transaction: VerifiedSignedTransaction,
+        signed_transaction: Option<VerifiedSignedTransaction>,
         locks_to_write: impl Iterator<Item = (ObjectRef, LockDetails)>,
     ) -> SuiResult {
         let mut batch = self.owned_object_locked_transactions.batch();
         batch.insert_batch(
             &self.owned_object_locked_transactions,
             locks_to_write.map(|(obj_ref, lock)| (obj_ref, LockDetailsWrapper::from(lock))),
         )?;
-        batch.insert_batch(
-            &self.signed_transactions,
-            std::iter::once((*transaction.digest(), transaction.serializable_ref())),
-        )?;
+        if let Some(signed_transaction) = signed_transaction {
+            batch.insert_batch(
+                &self.signed_transactions,
+                std::iter::once((
+                    *signed_transaction.digest(),
+                    signed_transaction.serializable_ref(),
+                )),
+            )?;
+        }
```

Comment:
Is my understanding correct that we'll no longer need for fastpath the `signed_transactions` anyways since we do not individually sign the transactions anymore? Looking on the read path this is mostly being used to tell via our APIs whether the transaction has been signed or not. Not sure if those APIs will be used in the future.

Reply:
Transactions will not be signed with BLS any more. But if there are use cases for `handle_transaction_info_request()` other than getting effects, we will have to figure out ways to migrate the existing users. I will leave a TODO over that API.